common/static/common/js/discussion/utils.js

```diff
@@ -291,21 +311,21 @@
             return errorsField.empty();
         };
 
-        DiscussionUtil.postMathJaxProcessor = function(text) {
+        DiscussionUtil.postMathJaxProcessor = function(htmlSnippet) {
             var RE_DISPLAYMATH, RE_INLINEMATH;
             RE_INLINEMATH = /^\$([^\$]*)\$/g;
             RE_DISPLAYMATH = /^\$\$([^\$]*)\$\$/g;
-            return this.processEachMathAndCode(text, function(s, type) {
+            return this.processEachMathAndCode(htmlSnippet, function(s, type) {
                 if (type === 'display') {
                     return s.replace(RE_DISPLAYMATH, function($0, $1) {
-                        return "\\[" + $1 + "\\]";
+                        return edx.HtmlUtils.HTML("\\[" + $1 + "\\]");
```

Comment:
Seems like the usage of the processor below assumes it is taking a string (that contains HTML) and would return a string (that contains HTML).  I think this code is converting to an HtmlSnippet, and then the concatenation below is just turning that back to a string.

I think I'd remove these 3 calls to `edx.HtmlUtisl.HTML`, unless I am missing something.